public/app/features/alerting/unified/components/rule-editor/GrafanaRuleInspector.tsx

```diff
@@ -0,0 +1,79 @@
+import { dump } from 'js-yaml';
+import React, { useEffect, useState } from 'react';
+import AutoSizer from 'react-virtualized-auto-sizer';
+
+import { CodeEditor, Drawer, useStyles2 } from '@grafana/ui';
+
+import { alertRuleApi } from '../../api/alertRuleApi';
+
+import { drawerStyles, RuleInspectorSubtitle, yamlTabStyle } from './RuleInspector';
+
+interface Props {
+  onClose: () => void;
+  alertUid: string;
+}
+
+export const GrafanaRuleInspector = ({ onClose, alertUid }: Props) => {
+  const [activeTab, setActiveTab] = useState('yaml');
+
+  const styles = useStyles2(drawerStyles);
+
+  return (
+    <Drawer
+      title="Inspect Alert rule"
+      subtitle={
+        <div className={styles.subtitle}>
+          <RuleInspectorSubtitle setActiveTab={setActiveTab} activeTab={activeTab} />
+        </div>
+      }
+      onClose={onClose}
+    >
+      {activeTab === 'yaml' && <GrafanaInspectorYamlTab alertUid={alertUid} />}
+    </Drawer>
+  );
+};
+
+const { useLazyExportRuleQuery } = alertRuleApi;
+
+interface YamlTabProps {
+  alertUid: string;
+}
+
+const GrafanaInspectorYamlTab = ({ alertUid }: YamlTabProps) => {
+  const styles = useStyles2(yamlTabStyle);
+
+  const [exportRule] = useLazyExportRuleQuery();
+
+  const [alertRuleAsYaml, setAlertRuleAsYaml] = useState('');
+
+  useEffect(() => {
+    exportRule({ uid: alertUid, format: 'yaml' })
```

Comment:
Wouldn't be simpler to use `useExportRuleQuery({ uid: alertUid, format: 'yaml' })` instead of this useEffect?

Then we wouldn't need `useState`. We could do like
```js
const yamlRule = useMemo(() => dump(exportRuleResult), [exportRuleResult]);
```